app/frontend/src/components/RatingsSlider/RatingsSlider.tsx

```diff
@@ -0,0 +1,142 @@
+import Slider from "@material-ui/core/Slider";
+import { withStyles } from "@material-ui/core/styles";
+import {
+  AMAZING,
+  NEGATIVE,
+  NEUTRAL,
+  POSITIVE,
+  RATINGS_SLIDER,
+} from "components/RatingsSlider/constants";
+import SliderLabel from "components/RatingsSlider/SliderLabel";
+import React, { useRef } from "react";
+import { theme } from "theme";
+import makeStyles from "utils/makeStyles";
+
+const StyledSlider = withStyles({
+  root: {
+    height: 8,
+    borderRadius: 25,
+    marginTop: theme.spacing(6),
+    marginBottom: theme.spacing(1),
+  },
+  track: {
+    height: 10,
+    borderRadius: 25,
+  },
+  rail: {
+    height: 10,
+    borderRadius: 25,
+  },
+  thumb: {
+    height: 20,
+    width: 20,
+    color: theme.palette.badRating.main,
+  },
+  valueLabel: {
+    left: "calc(-50% + 4px)",
+  },
+  mark: {
+    display: "none",
+  },
+  markLabel: {
+    maxWidth: "20%",
+    whiteSpace: "pre-line",
+  },
+})(Slider);
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    '& .MuiSlider-markLabel[data-index="0"]': {
+      transform: "translateX(0%)",
+    },
+    '& .MuiSlider-markLabel[data-index="3"]': {
+      transform: "translateX(10%)",
+    },
+  },
+}));
+
+const getSliderColor = (value: number) => {
+  const goodColor = theme.palette.goodRating.main.slice(1);
+  const badColor = theme.palette.badRating.main.slice(1);
+
+  const r = Math.ceil(
+    parseInt(goodColor.substring(0, 2), 16) * value +
+      parseInt(badColor.substring(0, 2), 16) * (1 - value)
+  );
+  const g = Math.ceil(
+    parseInt(goodColor.substring(2, 4), 16) * value +
+      parseInt(badColor.substring(2, 4), 16) * (1 - value)
+  );
+  const b = Math.ceil(
+    parseInt(goodColor.substring(4, 6), 16) * value +
+      parseInt(badColor.substring(4, 6), 16) * (1 - value)
+  );
+
+  const rgb = `rgb(${r}, ${g}, ${b})`;
+
+  return rgb;
+};
+
+const marks = [
+  {
+    value: 0,
+    label: `${NEGATIVE}`,
+  },
+  {
+    value: 0.33,
+    label: `${NEUTRAL}`,
+  },
+  {
+    value: 0.6,
+    label: `${POSITIVE}`,
+  },
+  {
+    value: 0.8,
+    label: `${AMAZING}`,
+  },
+];
+
+const handleSliderChange = (sliderRef: any, value: number | number[]) => {
```

Comment:
```suggestion
const handleSliderChange = (sliderRef: React.RefObject<HTMLSpanElement>, value: number | number[]) => {
```

Rather than ignoring the type error and typing it as `any` again, it's better to split the below out like so:
```typescript
const track = sliderRef.current.querySelector(".MuiSlider-track");
if (track) {
  track.style.backgroundColor = color;
}
```
etc

It's more tedious but much more type safe, as `querySelector` could return null if the CSS selector had a typo, or if MUI changes the class name in the next version. In both cases, we'd want to find out about these errors in compile time rather than crashing at run time with a can't access property X of `null` error which won't be good experience for users.

Reply:
Thank you! This with the small change of: 
```
const track = sliderRef.current.querySelector<HTMLSpanElement>(".MuiSlider-track");
...
```
finally did it.